### PR TITLE
[CL-273] Update styles for checkbox and form control

### PR DESCRIPTION
**Pull request by vleague2**
## 🎟️ Tracking

[CL-273](https://bitwarden.atlassian.net/browse/CL-273)

## 📔 Objective

This PR updates the styles for the checkbox component according to the new extension refresh design. It also impacts the radio button, which shares the parent `form-control` component. Radio buttons will be updated fully in a separate ticket/PR.

## 📸 Screenshots

Check Storybook Publish action > Form folder > Checkbox stories for full changes.

Screenshot 2024-07-19 at 10 08 01 AM

In the web app for standalone checkboxes:

Screenshot 2024-07-19 at 11 43 59 AM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[CL-273]: https://bitwarden.atlassian.net/browse/CL-273?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by willmartian — COMMENTED** [review]
Should the error/hint and label be aligned? cc @danielleflinn 

Screenshot 2024-07-23 at 10 54 36 AM

**Review by willmartian — CHANGES_REQUESTED** [review]
Since some of these css choices are to support standalone checkboxes and how they display in tables, could you add a story of a checkbox in a table row?

**Review by danielleflinn — CHANGES_REQUESTED** [review]
@willmartian yes the error should be aligned to the label just like the hint is. Also It looks like the error text is a bit large. Lets use the same font-size as the field hint/error here. 

+ 1 to Will's comment on the story for a checkbox in a table row. I also noticed while reviewing, the standalone stories don't have focus indicators. Can we add those so they display around the single checkbox?

File: libs/components/src/form-control/label.directive.ts [hunk 1]
```diff
@@ -7,10 +9,17 @@ let nextId = 0;
...
+    @Optional() private parentFormControl: FormControlComponent,
+  ) {}
 
   @HostBinding("class") @Input() get classList() {
-    return ["tw-truncate", "tw-inline-flex", "tw-gap-1", "tw-items-baseline", "tw-flex-row"];
+    const classes = ["tw-inline-flex", "tw-gap-1", "tw-items-baseline", "tw-flex-row"];
+    /**
+     * We don't want to truncate checkboxes or radio buttons, which use form-control
+     */
+    return this.parentFormControl ? classes : classes.concat(["tw-truncate"]);
```
Comment (willmartian) [inline libs/components/src/form-control/label.directive.ts:22]: Applying classes based on the presence of a parent can be accomplished with pure CSS/Tailwind:

`[&:not(bit-form-control_*)]:tw-truncate`

**Review by willmartian — APPROVED** [review]
Suggestion above is optional ^

**Pull request merged**